src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -794,39 +845,120 @@ private static GapicProductConfig createDummyInstance(
     ImmutableMap.Builder<String, ResourceNameConfig> resourceNameConfigs =
         new ImmutableSortedMap.Builder<>(Comparator.naturalOrder());
     resourceNameConfigs.putAll(finalSingleResourceNameConfigs);
-    resourceNameConfigs.putAll(fixedResourceNameConfigs);
+    resourceNameConfigs.putAll(finalFixedResourceNameConfigs);
     resourceNameConfigs.putAll(finalResourceOneofNameConfigs);
     return resourceNameConfigs.build();
   }
 
-  // Return map of fully qualified SingleResourceNameConfig name to its derived config.
+  /**
+   * Create all the ResourceNameOneofConfig from the protofile and GAPIC config, and let the GAPIC
+   * config resourceNames override the protofile resourceNames in event of clashes.
+   */
+  @Nullable
+  private static ImmutableMap<String, ResourceNameConfig>
+      createResourceNameConfigsForGapicConfigOnly(
+          DiagCollector diagCollector,
+          ConfigProto configProto,
+          @Nullable List<ProtoFile> protoFiles,
```

Comment:
These functions all take a protoFiles parameter, but (I think?) all of them use it to extract a single proto file, and nothing else. Should this parameter be a nullable single proto file?